File: 351002/Khmylko/rest-api/publisher/src/Controller/NoticeController.php
<?php
namespace App\Controller;

use App\Service\NoticeService;

class NoticeController {
    private NoticeService $service;

    public function __construct(NoticeService $service) {
        $this->service = $service;
    }

    public function getAll(): array {
        return $this->service->findAll();
    }

    public function getById(int $id): array {
        return $this->service->getById($id);
    }

    public function create(array $data): array {
        return $this->service->create($data);
    }

    public function update(int $id, array $data): array {
        return $this->service->update($id, $data);
    }

    public function delete(int $id): void {
        $this->service->delete($id);
    }
}

File: 351004/Shkredova/task310-rest/src/main/java/com/example/task310/controller/NoticeController.java
package com.example.task310.controller;

import com.example.task310.dto.NoticeRequestTo;
import com.example.task310.dto.NoticeResponseTo;
import com.example.task310.dto.kafka.NoticeMessage;
import com.example.task310.model.NoticeState;
import com.example.task310.service.NoticeKafkaService;
import com.example.task310.service.NoticeRestClientService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/notices")
@RequiredArgsConstructor
public class NoticeController {
    private final NoticeKafkaService noticeKafkaService;
    private final NoticeRestClientService noticeRestClientService;

    @PostMapping
    public ResponseEntity<NoticeResponseTo> create(@Valid @RequestBody NoticeRequestTo request) {
        NoticeMessage response = noticeKafkaService.sendForModeration(request.getNewsId(), request.getContent());
        NoticeResponseTo dto = new NoticeResponseTo();
        dto.setId(response.getId());
        dto.setNewsId(response.getNewsId());
        dto.setContent(response.getContent());
        // Преобразуем String в NoticeState
        dto.setState(NoticeState.valueOf(response.getState()));
        return ResponseEntity.status(HttpStatus.CREATED).body(dto);
    }

    @GetMapping
    public ResponseEntity<List<NoticeResponseTo>> findAll() {
        return ResponseEntity.ok(noticeRestClientService.findAll());
    }

    @GetMapping("/{id}")
    public ResponseEntity<NoticeResponseTo> findById(@PathVariable Long id) {
        return ResponseEntity.ok(noticeRestClientService.findById(id));
    }

    @PutMapping
    public ResponseEntity<NoticeResponseTo> update(@Valid @RequestBody NoticeRequestTo request) {
        return ResponseEntity.ok(noticeRestClientService.update(request));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        noticeRestClientService.delete(id);
        return ResponseEntity.noContent().build();
    }
}